### ISSUE TITLE
Unable to set datetime fields when mocking data

### ISSUE THREAD

**Issue by stevenhair** [cannatag/ldap3#342]
When trying to mock out datetime fields in entries, the following error occurs: `TypeError: 'datetime.datetime' object is not iterable`.

If a integer/string timestamp is used (e.g. `116303145750000000`), then the error does not occur, but the timestamp isn't converted to a Python datetime the same way that it does for live servers.

Example:

```python
>>> from datetime import datetime
>>> import ldap3
>>> server = ldap3.Server('ldap://0.0.0.0')
>>> connection = ldap3.Connection(server, client_strategy=ldap3.MOCK_SYNC)
>>> connection.strategy.add_entry(
...    'CN=Some User,OU=Users,DC=org,DC=com',
...    {
...        'sAMAccountName': 'some-user',
...        'lastLogon': [datetime(1969, 7, 21, 2, 56, 15)],
...    }
... )
True
>>> connection.bind()
True
>>> connection.search('DC=org,DC=com', '(sAMAccountName=some-user)', attributes=ldap3.ALL_ATTRIBUTES)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    connection.search('DC=org,DC=com', '(sAMAccountName=some-user)', attributes=ldap3.ALL_ATTRIBUTES)
  File "/Users/shair/.virtualenvs/web-auth/lib/python3.5/site-packages/ldap3/core/connection.py", line 766, in search
    response = self.post_send_search(self.send('searchRequest', request, controls))
  File "/Users/shair/.virtualenvs/web-auth/lib/python3.5/site-packages/ldap3/strategy/mockSync.py", line 69, in post_send_search
    response = search_result_entry_response_to_dict(entry, self.connection.server.schema, self.connection.server.custom_formatter, self.connection.c
heck_names)
  File "/Users/shair/.virtualenvs/web-auth/lib/python3.5/site-packages/ldap3/operation/search.py", line 522, in search_result_entry_response_to_dict
    entry['raw_attributes'] = raw_attributes_to_dict(response['attributes'])
  File "/Users/shair/.virtualenvs/web-auth/lib/python3.5/site-packages/ldap3/operation/search.py", line 414, in raw_attributes_to_dict
    attributes[str(attribute['type'])] = decode_raw_vals(attribute['vals'])
  File "/Users/shair/.virtualenvs/web-auth/lib/python3.5/site-packages/ldap3/operation/search.py", line 402, in decode_raw_vals
    return [bytes(val) for val in vals] if vals else None
  File "/Users/shair/.virtualenvs/web-auth/lib/python3.5/site-packages/ldap3/operation/search.py", line 402, in <listcomp>
    return [bytes(val) for val in vals] if vals else None
TypeError: 'datetime.datetime' object is not iterable
```

**Comment by cannatag** [comment]
If you want to pass Python objects to the mock strategy you must specify an offline schema. It seems to me that you're mocking an Active Directory server, so you must specify ``get_info=ldap3.OFFLINE_AD_2012_R2 to the Server definition.  Furthermore the Active directory timestamp is different from the standard LDAP generalized time format, so I added a "validator" for the lastLogon attribute. 

The strategy now also raises an exception if an entry is created with not bytes values when no offline schema is present.

Your code should be as following:

```
from datetime import datetime
import ldap3
from ldap3.core.timezone import OffsetTzInfo
server = ldap3.Server(' ldap://0.0.0.0', get_info=ldap3.OFFLINE_AD_2012_R2)
# server = ldap3.Server(' ldap://0.0.0.0')
connection = ldap3.Connection(server, client_strategy=ldap3.MOCK_SYNC)
connection.strategy.add_entry(
   'CN=Some User,OU=Users,DC=org,DC=com',
   {
       'sAMAccountName': 'some-user',
       # 'lastLogon': [datetime(1969, 7, 21, 2, 56, 15, tzinfo=OffsetTzInfo(0, 'abc'))],
       'lastLogon': [datetime(1969, 7, 21, 2, 56, 15)],
   }
)

connection.bind()

connection.search('DC=org,DC=com', '(sAMAccountName=some-user)', attributes=ldap3.ALL_ATTRIBUTES)
print(connection.entries)
```

Please get the code in dev and test it.

Bye,
Giovanni

**Comment by cannatag** [comment]
Released in ldap3 version 2.2.4

**Comment by stevenhair** [comment]
I'm sorry, I was pulled away on something else and this was my first chance to return to this. I can confirm that upgrading to 2.2.4 and adding `get_info` to the server definition solved my problem.

Thanks for your help! 🎉
